CONTRIBUTING.md

```diff
@@ -114,6 +123,50 @@ Examples of PR titles:
  - test: Add infinity test for `MoveEffect.to`
  - refactor: Optimize the structure of the game loop
 
+
+# Maintainers
+
+These instructions are for the maintainers of Flame.
+
+## Merging a pull request
+
+When merging a pull request, make sure that the title of the merge commit has the correct
+conventional commit tag and a descriptive title. This is extra important since sometimes the title
+of the PR doesn't reflect what GitHub defaults to for the merge commit title (if the title has been
+changed during the life time of the PR for example).
+
+All the default text should be removed from the commit message and the PR description and the
+instructions from the "Migration instruction" (if the PR is breaking) should be copied into the
+commit message.
+
+
+## Breaking changes
+
+When doing breaking changes a deprecation tag should be added when possible and contain a message
+that conveys to the user what which version that the deprecated method/field will be removed in and 
+what method they should use instead to perform the task. The version specified should be at least
+two versions after the current one, such that there will be at least one stable release where the
+users get to see the deprecation warning and in the version after that (or a later version) the
```

Comment:
The first part of this is general for all contributors imho. And we should maybe give an example

For example:

Lets say the current version is 1.1.2, and the next version that will have the deprecated function is 1.2.0, then the deprecated tag could look like this:

```dart
@Deprecated('Will be removed in 1.3.0')
void deprecatedFeature() {}
```

Reply:
That it already defined in the "Breaking changes" section, that section could be improved, but it's also outside the scope of this PR.

Reply:
Oh wait, that's this section. :sweat_smile: 
I thought we had this defined under "Open a pull request" too, this section should be moved to there.

Reply:
Moved it and added an example.